compiler/injectdestructors.nim

```diff
@@ -471,10 +472,9 @@ proc recurse(n: PNode, c: var Con, processProc: proc): PNode =
           branch.add processProc(n[i][0], c)
       result.add branch
   else:
-    assert(false, $n.kind)
+    result = copyTree(n)
 
 proc pExpr(n: PNode; c: var Con): PNode =
-  assert(isExpression(n), $n.kind)
```

Comment:
nkCall with a function that returns nothing or void `isExpression` returns false while nkCall is properly handled further on

Reply:
How does such a function end up here?

Reply:
Compile with --newruntime and --threads:on to trigger this assert:
```nim
import threadpool

proc test(a: string) = 
  echo a

spawn test("my")

```

Reply:
Can we add this as a test?